### PR TITLE
fix(span-tree): Allow expanding spans with multiple embedded transactions

### PR DESCRIPTION
**Pull request by 0Calories**
Fixes a bug where you could not expand spans with embedded transactions if there is more than one transaction within the same parent span. 

An example of a situation in where a span could have multiple embedded transactions is with NextJS: 
There is no common entrypoint between middleware and API routes, so they need to be created as separate transactions and attached on the client span that made the request. 

### Before
image

### After
image

### CODE REVIEW

**Review by lforst — COMMENTED** [review]
Just skimmed the code and it looks good to me! It's probably wise to have somebody else also look through it since I am not familiar with this part of the code base.

Would be sick to get this in!

File: static/app/components/events/interfaces/spans/spanTreeModel.tsx [hunk 1]
```diff
@@ -683,29 +685,44 @@ class SpanTreeModel {
...
+              removeTraceBounds(child.generateTraceBounds().spanId);
+            });
+          }
+        }
 
-      if (!this.showEmbeddedChildren) {
-        if (this.embeddedChildren.length > 0) {
+        if (this.showEmbeddedChildren) {
+          if (this.embeddedChildren.length === 0) {
+            Promise.all(
```
Comment (gggritso) [inline static/app/components/events/interfaces/spans/spanTreeModel.tsx:704]: I'm not familiar with Mobx, but does this need a `return`? There used to be a `return` here!

File: static/app/components/events/interfaces/spans/traceView.spec.tsx [hunk 2]
```diff
@@ -325,6 +325,163 @@ describe('TraceView', () => {
...
+            parent_event_id: '2b658a829a21496b87fd1f14a61abf65',
+            parent_span_id: 'b000000000000000',
+            project_id: project.id,
+            project_slug: project.slug,
+            span_id: 'aa5abb302ad5b9e1',
+            transaction:
+              '/api/0/organizations/{organization_slug}/events/{project_slug}:{event_id}/',
+            'transaction.duration': 159,
+            'transaction.op': 'middleware.nextjs',
+          },
```
Comment (gggritso) [inline static/app/components/events/interfaces/spans/traceView.spec.tsx:411]: These look the same as the mocks right above, right? If yes, would be nice to re-use!

**Review by gggritso — COMMENTED** [review]
Makes sense, just a couple of nits! Also would you mind adding a small note to the PR to explain why/when a span might have multiple linked transactions?

File: static/app/components/events/interfaces/spans/traceView.spec.tsx (hunk 2 above)
Comment (0Calories) [reply static/app/components/events/interfaces/spans/traceView.spec.tsx:411]: It's almost the same, the difference is that it has two embedded transactions set under the same parent as opposed to one. But you bring up a good point, I should make a const and then copy + modify it for this test to make that more clear

File: static/app/components/events/interfaces/spans/spanTreeModel.tsx (hunk 1 above)
Comment (0Calories) [reply static/app/components/events/interfaces/spans/spanTreeModel.tsx:704]: Good catch, I meant to return `Promise.all` here, although surprisingly it still works 🤔

**Comment by 0Calories** [comment]
Sorry @lforst, I've been running into some strange linting errors which have been blocking me on this PR. I'll try to fix this tomorrow morning

**Review by gggritso — APPROVED** [review]
Very nice!

**Pull request merged**